_renderers/vault.py

```diff
@@ -48,13 +48,13 @@ def _cached_read(path, cache_prefix='', **kwargs):
     if vault_data:
         log.debug('Loaded cached data for path %s', path)
         vault_data = vault_data['data']['value']
-        lease_start = datetime.strptime(vault_data['created'],
-                                        '%Y-%m-%dT%H:%M:%S.%f')
-        lease_length = timedelta(seconds=vault_data['lease_duration'])
-        # lease_info = vault_client.write('sys/leases/lookup',
-        #                                 lease_id=vault_data['lease_id'])
-        lease_valid = ((lease_start + lease_length) > (datetime.utcnow()
-                                        + timedelta(**renewal_threshold)))
+        lease = vault_client.get_lease(vault_data['lease_id'])
+
+        if (lease and timedelta(seconds=lease['data']['ttl']) >
```

Comment:
Why is the comparison using `lease and timedelta ...` and not just timedelta since we just need to figure out whether that's larger or small than the `renewal_threshold`

Reply:
Because we want to make sure that the lease actually exists before bothering to index into the structure and validate the TTL

Reply:
Because we want to make sure that the lease actually exists before bothering to index into the structure and validate the TTL